lib/pure/collections/sequtils.nim

```diff
@@ -434,7 +434,9 @@ proc delete*[T](s: var seq[T]; first, last: Natural) =
     var dest = @[1, 1, 1, 2, 2, 2, 2, 2, 2, 1, 1, 1, 1, 1]
     dest.delete(3, 8)
     assert outcome == dest
-
+  doAssert first <= last
```

Comment:
Previously sending `first > last` resulted in a `IndexError`, which doesn't tell the user much about what went wrong. If sending `first > last` is supposed to work I can fix it instead.

Reply:
IMO, first greater than last should silently delete nothing. This how you define empty sequence with inclusive ranges.

Reply:
And first >= len should be IndexError

Reply:
> And first >= len should be IndexError

I actually agree, but it would be inconsistent with how the `last` parameter is handled (out of bounds allowed) and how `system.delete` behaves. If `first >= len` results in IndexError then `last >= len` should also result in IndexError, which would be a breaking change.

Reply:
There is some Python influence here too, ``"oobasfd"[:10000] == 'oobasfd'``, Python is usually silent about index out of bounds for slices.